python/node_classification.py

```diff
@@ -0,0 +1,744 @@
+        if epoch % current_values[TrainParams.CHECKPOINT_FREQ] == 0:
+            if (
+                time() - last_time
+                > current_values[TrainParams.TIME_BETWEEN_CHECKPOINTS]
+            ):
+                save_model_to_folder()
+                last_time = time()
+
+    return [
+        mgp.Record(
+            epoch=data["epoch"],
+            loss=data["loss"],
+            val_loss=data["val_loss"],
+            train_log=data["train"],
+            val_log=data["val"],
+        )
+        for data in logged_data
+    ]
+
+
+##############################
+# get training data
+##############################
+
+
+@mgp.read_proc
+def get_training_data() -> mgp.Record(
+    epoch=int, loss=float, val_loss=float, train_log=mgp.Any, val_log=mgp.Any
+):
+    """This function is used so user can see what is logged data from training.
+
+
+    Returns:
+        mgp.Record(
+            epoch (int): epoch number of record of logged data row
+            loss (float): loss in logged data row
+            val_loss (float): validation loss in logged data row
+            train_log (mgp.Any): training parameters of record of logged data row
+            val_log (mgp.Any): validation parameters of record of logged data row
+            ): record to return
+
+
+    """
+
+    return [
+        mgp.Record(
+            epoch=logged_data[k]["epoch"],
+            loss=logged_data[k]["loss"],
+            val_loss=logged_data[k]["val_loss"],
+            train_log=logged_data[k]["train"],
+            val_log=logged_data[k]["val"],
+        )
+        for k in range(len(logged_data))
+    ]
+
+
+##############################
+# model loading and saving, predict
+##############################
+
+
+@mgp.read_proc
+def save_model() -> mgp.Record(path=str, status=str):
+    """This function saves model to model saving folder. If there are already total
+    of max_models_to_keep models in model saving folder, oldest model is deleted.
+
+    Exception: raised if model is not initialized or defined
+
+    Returns:
+        mgp.Record(
+            path (str): path to saved model
+            status (str): status of saving model
+            ): return record
+    """
+
+    if model is None:
+        raise Exception(
+            "Saving is not enabled until model is not initialized or loaded."
+        )
+
+    path_to_saved_model = save_model_to_folder()
+
+    return mgp.Record(
+        path=path_to_saved_model, status="Model has been successfully saved."
+    )
+
+
+@mgp.read_proc
+def load_model(ctx: mgp.ProcCtx, num: int = 0) -> mgp.Record(path=str, status=str):
+    """This function loads model from defined folder for saved models.
+
+    Args:
+        num (int, optional): ordinary number of model to load from default map. Defaults to 0 (newest model).
+
+    Returns:
+        mgp.Record(path (str): path to loaded model): return record
+    """
+    global model
+
+    data = declare_data(ctx)
+    _, _ = declare_model(data)
```

Comment:
```suggestion
    declare_model(data)
```
Alternatively, can this line be avoided?